src/stdlib_io.fypp

```diff
@@ -7,18 +7,45 @@ module stdlib_io
   !! ([Specification](../page/specs/stdlib_io.html))
 
   use stdlib_kinds, only: sp, dp, qp, &
-      int8, int16, int32, int64
+      int8, int16, int32, int64, lk, c_bool
   use stdlib_error, only: error_stop
   use stdlib_optval, only: optval
   use stdlib_ascii, only: is_blank
   implicit none
   private
   ! Public API
-  public :: loadtxt, savetxt, open
+  public :: loadtxt, savetxt, open, disp
 
   ! Private API that is exposed so that we can test it in tests
   public :: parse_mode
 
+  interface disp
+    !!
+    #:set DISP_KINDS_TYPES = REAL_KINDS_TYPES + INT_KINDS_TYPES &
+        & + CMPLX_KINDS_TYPES + LOG_KINDS_TYPES
+    #:set DISP_RANKS = range(0, 4)
+    #:for kind, type in DISP_KINDS_TYPES
+    #:for rank in DISP_RANKS
+    #:if rank != 3
```

Comment:
The preprocessor is quite complex here, since we are dealing with a range of 0, 1, 2, 3 where only rank 3 is special, maybe we can create two separate interface generation loops.

Reply:
Yes, because I consider that the screen and ASCII code text space are two-dimensional, and use `disp` to print up to three-dimensional arrays (one of the dimensions is realized by `do` loop), so the default highest dimension here is **three,** which has an upper limit . 
The existing `RANK` of `stdlib` seems to mainly serve the `stdlib_stats` routines. I took a look at it, and the dimensions inside generally need to be **three-dimensional or more**. 
But in fact, three-dimensional is not necessary, we can only leave this interface implementation:
```fortran
call disp( A(i, j, k, :, :) [, string] )    !! `i,j,k...` can be determined by `do` loop
```

Reply:
✔It is OK, I misunderstood what you meant at first, now I understand.👍